src/main/java/cubicchunks/world/column/CubeMap.java

```diff
@@ -68,12 +68,13 @@ Cube remove(int cubeY) {
 	 * @param cube the cube to add
 	 */
 	void put(@Nonnull Cube cube) {
+		// TODO trade memory for runtime? We could have an IntHashset storing added cube ids for faster contains()
+		// checks
```

Comment:
As in eliminating the getY()? I doubt a micro-optimization like this will give us any measurable performance gain.